File: 53_maximum_subarray/53_maximum_subarray.ts
function maxSubArray(nums: number[]): number {
  let sum: number = 0;
  let maxSum: number = Number.MIN_SAFE_INTEGER;

  for (let i: number = 0; i < nums.length; i++) {
    sum += nums[i];
    maxSum = Math.max(sum, maxSum);

    if (sum < 0) {
      sum = 0;
    }
  }
  return maxSum;
}

File: 53_maximum_subarray/53_maximum_subarray.go
package maximumsubarray

import "math"

func maxSubArray(nums []int) int {
	sum := 0
	maxSum := -math.MaxFloat64

	for _, val := range nums {
		sum += val
		maxSum = math.Max(float64(maxSum), float64(sum))

		if sum < 0 {
			sum = 0
		}
	}

	return int(maxSum)
}
